Sources/Twilio/Twilio.swift

```diff
@@ -1,80 +1,65 @@
-    func send(_ sms: OutgoingSMS, on container: Container) throws -> Future<Response>
+public protocol TwilioProvider {
+    func send(_ sms: OutgoingSMS) -> EventLoopFuture<ClientResponse>
 }
 
-// MARK: - Engine
 public struct Twilio: TwilioProvider {
-    public enum Error: Debuggable {
-        /// Encoding problem
-        case encodingProblem
-
-        /// Identifier
-        public var identifier: String {
-            switch self {
-            case .encodingProblem:
-                return "twilio.encoding_error"
-            }
-        }
-
-        /// Reason
-        public var reason: String {
-            switch self {
-            case .encodingProblem:
-                return "Encoding problem"
-            }
-        }
+    let application: Application
+    
+    public init (_ app: Application) {
+        application = app
     }
+}
 
-    /// Error response object
-    public struct ErrorResponse: Decodable {
+// MARK: - Configuration
 
-        /// Error messsage
-        public let message: String
+extension Twilio {
+    struct ConfigurationKey: StorageKey {
+        typealias Value = TwilioConfiguration
     }
 
-    public var accountId: String
-    public var accountSecret: String
-
-
-    /// Initializer
-    ///
-    /// - Parameters:
-    ///   - accountId: Account ID provided by Twilio
-    ///   - accountSecret: Account secret provided by Twilio
-    public init(accountId: String, accountSecret: String) {
-        self.accountId = accountId
-        self.accountSecret = accountSecret
+    public var configuration: TwilioConfiguration? {
+        get {
+            application.storage[ConfigurationKey.self]
+        }
+        nonmutating set {
+            application.storage[ConfigurationKey.self] = newValue
+        }
     }
+}
 
-    // MARK: Send message
+// MARK: Send message
 
+extension Twilio {
     /// Send sms
     ///
     /// - Parameters:
     ///   - content: outgoing sms
     ///   - container: Container
     /// - Returns: Future<Response>
-    public func send(_ sms: OutgoingSMS, on container: Container) throws -> Future<Response> {
-        let authKeyEncoded = try encode(accountId: self.accountId, accountSecret: self.accountSecret)
-
-        var headers = HTTPHeaders([])
-        headers.add(name: HTTPHeaderName.authorization, value: "Basic \(authKeyEncoded)")
-
-        let client: Client = try container.make()
-        return client.post(
-            "https://api.twilio.com/2010-04-01/Accounts/\(self.accountId)/Messages.json",
-            headers: headers
-        ) { request in
-            try request.content.encode(sms, as: MediaType.urlEncodedForm)
+    public func send(_ sms: OutgoingSMS) -> EventLoopFuture<ClientResponse> {
+        guard let configuration = self.configuration else {
+            fatalError("Twilio not configured. Use app.twilio.configuration = ...")
+        }
+        
+        return application.eventLoopGroup.future().flatMapThrowing { _ -> HTTPHeaders in
```

Comment:
Why is this wrapped as a `Future`? I don't think anything until `self.application.client.post` is async.

Reply:
Because
```swift
public func send(_ sms: OutgoingSMS) -> EventLoopFuture<ClientResponse> {
```
is not throwing anymore because in Vapor4 we use clean `NIO` and by `NIO` rules futures should not throw, but they should fail, so we should always wrap any throwing code into `flatMapThrowing`.